react/spring-page/client/src/components/Main/Main.js

```diff
@@ -0,0 +1,59 @@
+import React, { useState, useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import { Redirect } from 'react-router-dom';
+import './Main.css';
+import { Description } from '../Description/Description';
+import API from '../../utils/API';
+
+export function Main() {
+  const [name, setName] = useState('');
+  const [data, setData] = useState([]);
+  const isLoggedIn = useSelector((state) => state.isLogin);
+  useEffect(() => {
+    const fetchSearch = async () => {
+      const result = await API.get(`/info/spring?filter=${name}`, { headers: { Authorization: `Bearer ${localStorage.getItem('token')}` } });
+      setData(result.data);
+    };
+    fetchSearch();
+  }, [name]);
+
+  if (!isLoggedIn.isLogin && !localStorage.getItem('token')) {
```

Comment:
No need to check for token. Rather, if you get 401 for the server, you should log out user and remove token (read about Axios interceptors)